validator/valnode/redis/consumer.go

```diff
@@ -163,10 +163,19 @@ func (s *ValidationServer) Start(ctx_in context.Context) {
 				res, err := valRun.Await(ctx)
 				if err != nil {
 					log.Error("Error validating", "request value", work.req.Value, "error", err)
+					err := s.consumers[work.moduleRoot].SetError(ctx, work.req.ID, err.Error())
 					work.req.Ack()
+					if err != nil {
+						log.Error("Error setting error for request", "id", work.req.ID, "error", err)
+					}
 				} else {
 					log.Debug("done work", "thread", i, "workid", work.req.ID)
 					err := s.consumers[work.moduleRoot].SetResult(ctx, work.req.ID, res)
+					if err != nil {
+						log.Error("Error setting result for request", "id", work.req.ID, "result", res, "error", err)
+					}
+					// Set an empty error even if the result is set successfully, as the error key is always checked first.
+					err = s.consumers[work.moduleRoot].SetError(ctx, work.req.ID, "")
```

Comment:
what is the motivation behind always setting the error response? Couldn't we skip that?

Reply:
Done.